test/fixtures_isce3.py

```diff
@@ -0,0 +1,165 @@
+
+
+@fixture(scope=determine_scope)
+def isce3_build_tag() -> str:
+    """Return a tag for images in the ISCE3 build chain."""
+    return generate_tag("isce3-setup")
+
+
+@fixture(scope=determine_scope)
+def isce3_setup_images(
+    isce3_build_tag: str,
+    base_tag: str,
+    cuda_version: Tuple[int, int],
+    cuda_repo_ver: str,
+    runtime_lockfile_path: Path,
+    dev_lockfile_path: Path,
+) -> Iterator[dict[str, Image]]:
+    """Return a dictionary of ISCE3 setup images."""
+    cuda_major, cuda_minor = cuda_version
+
+    image_dict = setup_all(
+        base=base_tag,
+        tag=isce3_build_tag,
+        no_cache=False,
+        cuda_version=f"{cuda_major}.{cuda_minor}",
+        cuda_repo=cuda_repo_ver,
+        runtime_env_file=runtime_lockfile_path,
+        dev_env_file=dev_lockfile_path,
+        verbose=True,
+        test=True,
+    )
+
+    yield image_dict
+
+    for tag in image_dict:
+        remove_docker_image(tag)
+
+
+@fixture(scope=determine_scope)
+def isce3_env_dev_image_tag(
+    isce3_build_tag: str,
+    isce3_setup_images: dict[str, Image],
+) -> str:
+    """Return the tag of the ISCE3 development environment image."""
+    pattern = re.compile(rf".*{isce3_build_tag}.*mamba-dev")
+    for tag in isce3_setup_images:
+        if pattern.match(tag):
+            return tag
+
+    raise ValueError("No development environment image tag found.")
+
+
+@fixture(scope=determine_scope)
+def isce3_git_repo_tag() -> str:
+    """Return a tag for the ISCE3 repository image."""
+    return generate_tag("isce3-repo")
+
+
+@fixture(scope=determine_scope)
+def isce3_git_repo_image(
+    isce3_env_dev_image_tag: str,
+    isce3_git_repo_tag: str,
+    base_properties: Tuple[Any, URLReader],
+) -> Iterator[Image]:
+    """Return a tag for the ISCE3 repository image."""
+    archive = "https://github.com/isce-framework/isce3/archive/refs/tags/v0.16.0.tar.gz"
+    _, url_reader = base_properties
+
+    dockerfile = git_extract_dockerfile(
+        base=isce3_env_dev_image_tag,
+        archive_url=archive,
+        directory=Path("/src/"),
+        url_reader=url_reader,
+    )
+
+    yield Image.build(
+        tag=isce3_git_repo_tag, dockerfile_string=dockerfile, no_cache=False
+    )
+
+    remove_docker_image(isce3_git_repo_image)
+
+
+@fixture(scope=determine_scope)
+def isce3_cmake_config_tag() -> str:
+    """Return a tag for the ISCE3 CMake config image."""
+    return generate_tag("isce3-cmake-config")
+
+
+@fixture(scope=determine_scope)
+def isce3_cmake_config_image(
+    isce3_cmake_config_tag: str,
+    isce3_git_repo_tag: str,
+    isce3_git_repo_image: Image,  # type: ignore
+) -> Iterator[Image]:
+    """Return the ISCE3 CMake config image."""
+
+    dockerfile = cmake_config_dockerfile(
+        base=isce3_git_repo_tag,
+        build_type="Release",
+        with_cuda=True,
+    )
+
+    yield Image.build(
+        tag=isce3_cmake_config_tag, dockerfile_string=dockerfile, no_cache=True
```

Comment:
Should we use the cache here as well?

```suggestion
        tag=isce3_cmake_config_tag, dockerfile_string=dockerfile, no_cache=False
```